src/commands/mod.rs

```diff
@@ -172,7 +177,21 @@ enum Command {
     Tag(tag::Opts),
 }
 
+#[cfg_attr(feature = "completions", allow(unused))]
 pub async fn execute() -> Result<()> {
+    #[cfg(feature = "completions")]
+    {
+        // Generate completions if running with "completions" feature enabled
+        //
+        // A bit hacky. Including source code of module in build.rs not work
+        // because there are more one file with clap::Command declaration
+        //
```

Comment:
I don't know if clap with subcommands implementing `clap::Command` is the right way. If not, we should of course change this.

Actually, I like the "hack" with adding the feature and then using the built binary to produce the completions. What about adding a standard subcommand `generate` which is only activated when activating the feature `completions`? Then the code to actually generate the completions could go into `src/commands/generate.rs` and if anyone wants to distribute a version which is able to generate subcommands can compile with the feature selected.

Reply:
Thought about command, but name `completions`, like in [poetry](https://python-poetry.org/docs/#enable-tab-completion-for-bash-fish-or-zsh)